### PR TITLE
Remove Deprecated EmailValidator

### LINKED ISSUE #1794: Replace E-mail Validator
Our current e-mail validator is not good. It is a simplistic regular expression that incorrectly flags some e-mails as being valid though they are not.

The [validates_email_format_of](https://github.com/validates-email-format-of/validates_email_format_of) gem can be of assistance here.  It has a superior implementation of e-mail validation, and superior test cases.

We should consider using this solution rather than our current validator.

### PR DESCRIPTION
**Pull request by JDutil**
**Description**

The email validator was deprecated Mar 15th 2018 almost 18 months ago
as part of: 6d8fd3c

The deprecation warning was part of v2.6.0rc1, and remains to this day
in v2.10.0.beta1 I think it's time to remove it altogether. It's been
4 minor version upgrades now.

Fixes #1794

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have updated Guides and README accordingly to this change (if needed)
- [x] I have added tests to cover this change (if needed)

### CODE REVIEW

**Comment by spaghetticode** [comment]
While I agree it's ok to remove `EmailValidator`, I'm not sure about completely removing the email validator `Spree::EmailValidator` as well. 

I think a more conservative approach may be to deprecate `Spree::EmailValidator` first, as somebody may be using it, and remove it with a later version of Solidus.

File: core/lib/spree/core/validators/email.rb [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-    def validate_each(record, attribute, value)
-      unless value =~ EMAIL_REGEXP
-        record.errors.add(attribute, :invalid, { value: value }.merge!(options))
-      end
-    end
-  end
-end
-
-# @private
-EmailValidator = ActiveSupport::Deprecation::DeprecatedConstantProxy.new(
```
Comment (kennyadsl) [inline core/lib/spree/core/validators/email.rb]: This is the only part to remove if we want to get rid of the deprecation message.

Comment (JDutil) [reply core/lib/spree/core/validators/email.rb]: Yeah makes sense I didn't notice that it was a proxy 👍

**Comment by JDutil** [comment]
@kennyadsl updated 👍

**Comment by kennyadsl** [comment]
I think specs are failing due to the `core/lib/spree/core/validators/email.rb → core/lib/spree/validators/email.rb` rename, which should not happen. Also, can you please remove the changelog entry? It will be autogenerated using the PR title, so it's better to change it and remove the entry, thanks!

**Comment by JDutil** [comment]
@kennyadsl updated 👍

**Pull request merged**